### PR TITLE
DAOS-10743 dtx: test case for handle DTX commit failure

### PR DESCRIPTION
**Pull request by Nasf-Fan**
New fail_loc to simulate the case that one of DTX participants failed
to commit some DTX locally, under such case, as long as one of others
commit the DTX successfully, such DTX will not be aborted.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

**Review by NiuYawei — DISMISSED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
Rebased because of merge conflict.

**Review by NiuYawei — APPROVED** [review]

File: src/dtx/dtx_rpc.c [hunk 1]
```diff
@@ -131,7 +131,7 @@ dtx_req_cb(const struct crt_cb_info *cb_info)
 
 	dout = crt_reply_get(req);
 	if (dra->dra_opc == DTX_COMMIT) {
-		dra->dra_committed += dout->do_misc;
+		*dra->dra_committed += dout->do_misc;
```
Comment (jolivier23) [inline src/dtx/dtx_rpc.c:134]: minor nit, I really prefer using parenthesis.  Otherwise it makes my brain hurt figuring out what we are dereferencing.

**Review by jolivier23 — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

File: src/dtx/dtx_rpc.c (hunk 1 above)
Comment (Nasf-Fan) [reply src/dtx/dtx_rpc.c:134]: sure, will fix it when refresh the patch.

**Review by NiuYawei — DISMISSED** [review]

File: src/dtx/dtx_rpc.c (hunk 1 above)
Comment (NiuYawei) [reply src/dtx/dtx_rpc.c:134]: (*dra->dra_committed) += dout->do_misc could be safer.

**Review by NiuYawei — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**